shadows/framework/src/main/java/org/robolectric/shadows/ImageUtil.java

```diff
@@ -102,16 +103,16 @@ public static boolean writeToStream(
         ImageWriteParam iwparam = writer.getDefaultWriteParam();
         iwparam.setCompressionMode(ImageWriteParam.MODE_EXPLICIT);
         iwparam.setCompressionQuality((quality / 100f));
+        int width = realBitmap.getWidth();
+        int height = realBitmap.getHeight();
         BufferedImage bufferedImage =
             new BufferedImage(
-                realBitmap.getWidth(),
-                realBitmap.getHeight(),
+                width,
+                height,
                 getBufferedImageType(realBitmap.getConfig(), needAlphaChannel(format)));
-        for (int x = 0; x < realBitmap.getWidth(); x++) {
-          for (int y = 0; y < realBitmap.getHeight(); y++) {
-            bufferedImage.setRGB(x, y, realBitmap.getPixel(x, y));
-          }
-        }
+        ShadowBitmap shadowBitmap = Shadow.extract(realBitmap);
+        int[] pixels = shadowBitmap.getPixelsInternal();
```

Comment:
nit: this needs a null check. In Robolectric you can create bitmpats with null colors using Shadow.newInstanceOf(Bitmap.class) and probably other methods :(


```
        if (pixels != null) {
          bufferedImage.setRGB(0, 0, width, height, pixels, 0, width);
        }
```